Refactor quant levels visualization

asl3 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #82790

### Summary
Refactors quantization levels visualization function to include alpha qparam in parameters of `float_to_apot` function call (due to `float_to_apot` function update). Also adds additional detail to the documentation for `quant_levels_visualization`.

### Test Plan
Print visualization by calling `quant_levels_visualization` function.

asl3 committed:
Refactor quant levels visualization

[ghstack-poisoned]

jerryzh168 commented on torch/ao/quantization/experimental/observer.py:
```diff
@@ -133,10 +136,16 @@ def forward(self, x_orig):
+            quantization_levels: quantization levels APoT qparam (Tensor)
+            level_indices: level indices APoT qparam (Tensor)
+    """
+    def quant_levels_visualization(self, alpha, quantization_levels, level_indices):
```
are these arguments calculated from `calculate_qparams`? maybe we can simplify this method by calling `calculate_qparams` to get these arguments and then do the visualization?

jerryzh168 commented on torch/ao/quantization/experimental/observer.py:
```diff
@@ -133,10 +136,16 @@ def forward(self, x_orig):
+            quantization_levels: quantization levels APoT qparam (Tensor)
+            level_indices: level indices APoT qparam (Tensor)
+    """
+    def quant_levels_visualization(self, alpha, quantization_levels, level_indices):
```
maybe there is some misunderstanding, what I meant is not to have these args, but get them from calculate_qparams:
```
alpha, gamma, quantization_levels, level_indices = self.calculate_qparams(...)
```

asl3 replied:
sorry i made that change but accidentally didn't include in my last push, should be correct now!

asl3 committed:
Update on "Refactor quant levels visualization"


### Summary
Refactors quantization levels visualization function to include alpha qparam in parameters of `float_to_apot` function call (due to `float_to_apot` function update). Also adds additional detail to the documentation for `quant_levels_visualization`.

### Test Plan
Print visualization by calling `quant_levels_visualization` function.

[ghstack-poisoned]

asl3 committed:
Update on "Refactor quant levels visualization"


### Summary
Refactors quantization levels visualization function to include alpha qparam in parameters of `float_to_apot` function call (due to `float_to_apot` function update). Also adds additional detail to the documentation for `quant_levels_visualization`.

### Test Plan
Print visualization by calling `quant_levels_visualization` function.

[ghstack-poisoned]

asl3 commented:
@pytorchbot merge -g
